### DISCUSSION TITLE
extend-include farkı

### DISCUSSION THREAD

**Discussion by ipekozbay** [19/classroom#50] [Ruby] [answered]
Extend ve include farkını anlayamadım.
asagıdaki kodda extend neden yazamadım ya da neden include yazdım açıklayabilirseniz sevinirim.teşekkürler 
<img width="231" alt="Ekran Resmi 2022-06-29 19 05 01" src="https://user-images.githubusercontent.com/72815766/176483397-3588b280-cd1b-4a10-8421-37d79882c136.png">

**Comment by roktas** [comment, marked as answer]
Çok temel bir konuyu sormuşsun, ders kayıtlarını seyrettin mi?

`class C; include M; end` ile `M` modülünün metotları (daha teknik söylenecek olursa `M` modülünün "instance" metotları) `C` sınıfının nesne metotları haline gelir.  `class C; extend M; end` ile `M` modülünün metotları `C` sınıfının sınıf metotları haline gelir.  Bu tanımda geçen "nesne metotu" ne demek oluyor, "sınıf metotu" ne demek oluyor; anladığını varsayıyorum.  Anlamadıysan temelde çok sorun vardır ve bu temeli bir an önce almalısın.  Daima noktanın soluna bakacaksın.  Bir metot inşa edilen bir nesne üzerinden çalıştırılıyorsa (örnekte `a`) bu bir nesne metotudur (`ayaz`); sınıf metotları ise (adı üzerinde) sınıflar üzerinden çalıştırılır, yani noktanın solunda bir sınıf vardır (ör. `A.ayaz`).

**Reply by ipekozbay** [reply]
Ders kayıtlarında ses olmamasından dolayı izleyemedim hocam sitenizdeki slaytlar üzerinden çalışıyorum. Extend include farkını anladım şu an teşekkür ederim.
